src/Criteria.php

```diff
@@ -155,25 +155,34 @@ public function getWhereExpression()
      */
     public function getOrderings()
     {
-        return $this->orderings;
+        return array_map(
+            static fn (Order $ordering): string => $ordering->value,
+            $this->orderings,
+        );
     }
 
     /**
      * Sets the ordering of the result of this Criteria.
      *
      * Keys are field and values are the order, being either ASC or DESC.
      *
-     * @see Criteria::ASC
-     * @see Criteria::DESC
+     * @see Order::ASC
+     * @see Order::DESC
      *
-     * @param array<string, string> $orderings
+     * @param array<string, string|Order> $orderings
      *
      * @return $this
      */
     public function orderBy(array $orderings)
     {
         $this->orderings = array_map(
-            static fn (string $ordering): string => strtoupper($ordering) === self::ASC ? self::ASC : self::DESC,
+            static function (string|Order $ordering): Order {
+                if ($ordering instanceof Order) {
+                    return $ordering;
+                }
+
+                return strtoupper($ordering) === Order::ASC->value ? Order::ASC : Order::DESC;
```

Comment:
Trigger a deprecation?

Reply:
Should we give the ORM a chance to upgrade first?

Reply:
I don't think we need to wait for that.

Reply:
Ok, let me add that deprecation layer then.